src/reducers/playlist.tsx

```diff
@@ -0,0 +1,37 @@
+import { find } from 'lodash';
+import { combineReducers, Reducer } from 'redux';
+import * as UrlPattern from 'url-pattern';
+import { RootAction } from '../actions';
+import * as action_types from '../actions/actionTypes';
+import { AppState, PlaylistInstanceState, PlaylistState } from '../types/store';
+
+
+const PLAYLIST_PATTERN = new UrlPattern('/playlist/(:playlistID)(.*)');
+
+
+const all: Reducer<PlaylistInstanceState[], RootAction> = (state = [], action) => {
+  switch (action.type) {
+    case action_types.NEW_PLAYLIST:
+      return state.concat(action.payload.playlist);
+    default:
+      return state;
+  }
+};
+
+
+export const playlist = combineReducers<PlaylistState, RootAction>({
+  all,
+});
+
+
+export const getActivePlaylist = (state: AppState) => {
+  const match = PLAYLIST_PATTERN.match(state.router.location.pathname);
```

Comment:
Not sure I understand this part.
Do you assume that `getActivePlaylist` is called only when the playlist is part of the URL? Is that really the only case?